internal/services/appservice/app_active_slot_common.go

```diff
@@ -0,0 +1,178 @@
+package appservice
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/web/mgmt/2021-02-01/web"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ActiveSlotResource struct{}
+
+type ActiveSlotModel struct {
+	SlotID              string `tfschema:"slot_id"`
+	OverwriteNetworking bool   `tfschema:"overwrite_network_config"` // Note: This setting controls the ambiguously named `PreserveVnet`
+	LastSwap            string `tfschema:"last_successful_swap"`
+}
+
+func (r ActiveSlotResource) ModelObject() interface{} {
+	return ActiveSlotModel{}
+}
+
+func (r ActiveSlotResource) ResourceType() string {
+	return "" // This is never called
+}
+
+var _ sdk.ResourceWithUpdate = ActiveSlotResource{}
+
+func (r ActiveSlotResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return nil // This is never called
+}
+
+func (r ActiveSlotResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"slot_id": {
+			Type:        pluginsdk.TypeString,
+			Required:    true,
+			Description: "The ID of the Slot to swap with `Production`.",
+			ValidateFunc: validation.Any(
+				validate.WebAppSlotID,
+				validate.FunctionAppSlotID,
+			),
+		},
```

Comment:
since this allows either but the parse function doesn't below - should we be overriding the arguments function and providing the validation function as an argument to this base function?